core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -0,0 +1,181 @@
+package org.ensime.core
+
+import scala.concurrent.duration._
+
+import akka.actor._
+import akka.event.LoggingReceive.withLabel
+import org.ensime.api._
+import org.ensime.config.richconfig._
+import org.ensime.util.{ Debouncer, Timing }
+import org.ensime.util.FileUtils.toSourceFileInfo
+import org.ensime.util.file._
+
+final case object SuspendAnalyzer
+
+class AnalyzerManager(
+    broadcaster: ActorRef,
+    analyzerCreator: List[EnsimeProjectId] => Props,
+    implicit val config: EnsimeConfig
+) extends Actor with ActorLogging with Stash {
+
+  // is this okay ?
+  private val suspendAnalyzer = Debouncer.forActor(
```

Comment:
you don't want a debouncer. You want to a `scheduler` to send the `Suspend` message in 5 minutes, but cancel it every time a message comes in.